lib/study.js

```diff
@@ -0,0 +1,54 @@
+const self = require('sdk/self');
+const shield = require('shield-studies-addon-utils');
+const { when: unload } = require('sdk/system/unload');
+
+const { Advisor } = require('./Advisor');
+
+let Advise = null;
+
+const studyConfig = {
+  name: self.addonId,
+  duration: 7,
+  surveyUrls: {
+    'end-of-study': 'mozilla.org',
+    'user-ended-study': 'mozilla.org',
+    ineligible: null,
+  },
+  variations: {
+    'passive-panel'() {
+      Advise.start('passivePanel');
+    },
+    'auto-open-panel'() {
+      Advise.start('autoOpenPanel');
+    },
+    'observe-only'() {},
+  },
+};
+
+class SecurityAdvisorStudy extends shield.Study {
+  constructor(config) {
+    super(config);
+    Advise = new Advisor();
+  }
+  isEligible() {
+    return super.isEligible() && Advise.isEligible();
+  }
+  cleanup() {
+    if (this.variation !== 'observe-only') {
```

Comment:
I think you're right that it's safe to run the `Advise.cleanup()` method for the `observe-only` branch. Previously it was necessary because in the `observe-only` branch, it was always the case that `Advise = null`. That's changed now with the `Advise.start()` method that the active branches use, so I'll remove the if-check.